packages/connection-storage/src/provider.ts

```diff
@@ -0,0 +1,44 @@
+import { createContext, createElement, useMemo, useContext } from 'react';
+import { ConnectionRepository } from './connection-repository';
+import type { ConnectionStorage } from './renderer';
+
+export const ConnectionStorageContext = createContext<
+  typeof ConnectionStorage | null
```

Comment:
You probably want the ConnectionStorageFacade here?

Reply:
Not necessarily, this provides a ConnectionStorage that will be also used for import-export, and we are not changing it.

Reply:
Interesting, I was worried that we might be passing an incomplete type here that would  become problematic for ConnectionStorage or ConnectionRepository but I just re-read and it looks alright because ConnectionRepository relies on few small picks from `typeof ConnectionStorage`.